sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -221,7 +221,7 @@ static void CleanupThread()
       // removed.
       for (auto connection = --(index->second.end());
            index->second.size() > 0 && connection->get()->IsExpired();
-           connection--)
+           connection = index->second.size() > 0 ? --connection : connection)
```

Comment:
I am confused as to why this was required just as a fix for windows? Why didn't it fail on linux?
Can you add a comment as well as to why we need to only decrement the connection with a prefix decrement, and only when size() > 0? Is there some instruction reordering happening by the compiler?

https://dev.azure.com/azure-sdk/public/_build/results?buildId=846649&view=logs&j=becb4c78-6d46-5b40-696d-491037a3748a&t=7da4ec77-80ef-5cae-b212-17e17d7c0a17

Reply:
Msvc adds a precondition to avoid decreasing an iterator below .begin().   So, if we do
auto it = someVector.begi();
it--;   //  This makes application crash on MSVC with a precondition where we can't go below the begin().
 
But on g++ and clang it works.

Reply:
This loop starts on `.end() - 1 ` position.  On each loop,  the position can be moved to `.end()` if the item is removed.
If the position goes to `.end()` and the list becomes empty, trying to go to `end() - 1` triggers MSVC precondition to abort